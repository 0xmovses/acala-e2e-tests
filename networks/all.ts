import { Config } from './types'

import polkadotConfig from './polkadot'
import collectivesConfig from './collectives'
import assetHubConfig from './asset-hub'

const all = {
  polkadot: polkadotConfig,
  assetHub: assetHubConfig,
  collectives: collectivesConfig,
} satisfies Record<string, Config>

export default all
