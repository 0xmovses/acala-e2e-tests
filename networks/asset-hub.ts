import { Config } from './types'

export default {
  polkadot: {
    name: 'polkadot-asset-hub' as const,
    endpoint: 'wss://polkadot-asset-hub-rpc.polkadot.io',
  },
  kusama: {
    name: 'kusama-asset-hub' as const,
    endpoint: 'wss://kusama-asset-hub-rpc.polkadot.io',
  },
  config: ({ alice }) => ({
    storages: {
      System: {
        Account: [[[alice.address], { providers: 1, data: { free: 10 * 1e12 } }]],
      },
    },
  }),
} satisfies Config

export const kusamaRelay = {
  paraId: 1000,
  ksm: { Token: 'KSM'}
} as const

export const assetHubKusamaPara= {
  paraId: 1000,
 ksm: { Token: 'KSM' }
} as const
