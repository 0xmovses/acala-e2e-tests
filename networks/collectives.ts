import { Config } from './types'

export default {
  relaychain: {
    name: 'polkadot' as const,
    endpoint: 'wss://rpc.polkadot.io',
  },
  parachain: {
    name: 'collectives' as const,
    endpoint: 'wss://polkadot-collectives-rpc.polkadot.io',
  },
  config: ({ alice }) => ({
    storages: {
      System: {
        Account: [[[alice.address], { providers: 1, data: { free: 10 * 1e12 } }]],
      },
    },
  }),
} satisfies Config

export const kusamaRelay = {
  paraId: 1000,
  ksm: { Token: 'KSM'}
} as const


export const collectives = {
  paraId: 2000,
  paraAccount: '13YMK2eYoAvStnzReuxBjMrAvPXmmdsURwZvc62PrdXimbNy',
  dot: { Token: 'DOT' },
  ksm: { Token: 'KSM' },
} as const