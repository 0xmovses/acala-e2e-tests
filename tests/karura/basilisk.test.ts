import { afterAll, beforeEach, describe, expect, it } from 'vitest'
import { connectParachains } from '@acala-network/chopsticks'

import { expectEvent, expectExtrinsicSuccess, expectJson, sendTransaction, testingPairs } from '../helper'
import networks from '../networks'

describe('Karura <-> Basilisk', async () => {
  const basilisk = await networks.basilisk()
  const karura = await networks.karura()
  await connectParachains([basilisk.chain, karura.chain])

  const { alice } = testingPairs()

  afterAll(async () => {
    await basilisk.teardown()
    await karura.teardown()
  })

  beforeEach(async () => {
    await karura.dev.setStorage({
      System: {
        Account: [[[alice.address], { data: { free: 10 * 1e12 } }]]
      },
      Tokens: {
        Accounts: [
          [[alice.address, { Token: 'KSM' }], { free: 10 * 1e12 }],
          [[alice.address, { Token: 'AUSD' }], { free: 10 * 1e12 }]
        ]
      },
      Sudo: {
        Key: alice.address
      }
    })
    await basilisk.dev.setStorage({
      System: {
        Account: [[[alice.address], { data: { free: 10 * 1e12 } }]]
      }
    })
  })

  it('0. Karura transfer DAI to basilisk', async () => {
    const DAI = "0x4bb6afb5fa2b07a5d1c499e1c3ddb5a15e709a71"
    const tx0 = await sendTransaction(
      karura.api.tx.sudo.sudoAs("rPWzRkpPjuceq6Po91sfHLZJ9wo6wzx4PAdjUH91ckv81nv", karura.api.tx.currencies.transfer(alice.address, { Erc20: DAI }, "1000000000000000000"))
        .signAsync(alice))

    await karura.chain.newBlock()
    expectExtrinsicSuccess(await tx0.events)

    console.dir((await tx0.events).map(x => x.toHuman()), { depth: null })
    expectEvent(await tx0.events, {
      event: expect.objectContaining({
        section: 'evmAccounts',
        method: 'ClaimAccount'
      })
    })
    //
    expectEvent(await tx0.events, {
      event: expect.objectContaining({
        section: 'currencies',
        method: 'Transferred'
      })
    })
    //
    // const tx1 = await sendTransaction(
    //   karura.api.tx.xTokens
    //     .transfer(
    //       {
    //         Erc20: DAI
    //       },
    //       10 ** 18,
    //       {
    //         V1: {
    //           parents: 1,
    //           interior: {
    //             X2: [
    //               {
    //                 Parachain: '2090'
    //               },
    //               {
    //                 AccountId32: {
    //                   network: 'Any',
    //                   id: alice.addressRaw
    //                 }
    //               }
    //             ]
    //           }
    //         }
    //       },
    //       'Unlimited'
    //     )
    //     .signAsync(alice)
    // )
    //
    // await karura.chain.newBlock()
    // await basilisk.chain.upcomingBlock()
    // expectExtrinsicSuccess(await tx1.events)

  })
})
