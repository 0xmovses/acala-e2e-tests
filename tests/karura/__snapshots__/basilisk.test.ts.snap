// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Karura <-> Basilisk > Karura transfer USDC to basilisk > events 1`] = `
[
  {
    "data": {
      "accountId": "rPizfonc8MPuEsMxufAytHjpRW7a2YrXWHKibaJXe9ZtjKx",
      "evmAddress": "0x4d9cf56ac6f1b4b6be4757ad0af3348fc6a918c2",
    },
    "method": "ClaimAccount",
    "section": "evmAccounts",
  },
  {
    "data": {
      "amount": "(redacted)",
      "currencyId": {
        "Erc20": "0x1f3a10587a20114ea25ba1b388ee2dd4a337ce27",
      },
      "from": "r7ts9D2xjiWoPhdnSLe3id9MAT5MADP8bBg17zP9aqRcKvj",
      "to": "rPizfonc8MPuEsMxufAytHjpRW7a2YrXWHKibaJXe9ZtjKx",
    },
    "method": "Transferred",
    "section": "currencies",
  },
]
`;

exports[`Karura <-> Basilisk > Karura transfer USDC to basilisk > events 2`] = `
[
  {
    "data": {
      "contract": "0x1f3a10587a20114ea25ba1b388ee2dd4a337ce27",
      "from": "0x4d9cf56ac6f1b4b6be4757ad0af3348fc6a918c2",
      "logs": [
        {
          "address": "0x1f3a10587a20114ea25ba1b388ee2dd4a337ce27",
          "data": "(redacted)",
          "topics": [
            "(redacted)",
            "(redacted)",
            "(redacted)",
          ],
        },
      ],
      "usedGas": "(redacted)",
      "usedStorage": "-64",
    },
    "method": "Executed",
    "section": "evm",
  },
  {
    "data": {
      "messageHash": "(redacted)",
    },
    "method": "XcmpMessageSent",
    "section": "xcmpQueue",
  },
  {
    "data": {
      "assets": [
        {
          "fun": {
            "Fungible": "(redacted)",
          },
          "id": {
            "Concrete": {
              "interior": {
                "X2": [
                  {
                    "Parachain": "(redacted)",
                  },
                  {
                    "GeneralKey": "0x021f3a10587a20114ea25ba1b388ee2dd4a337ce27",
                  },
                ],
              },
              "parents": "(redacted)",
            },
          },
        },
      ],
      "dest": {
        "interior": {
          "X2": [
            {
              "Parachain": "(redacted)",
            },
            {
              "AccountId32": {
                "id": "(redacted)",
                "network": "Any",
              },
            },
          ],
        },
        "parents": "(redacted)",
      },
      "fee": {
        "fun": {
          "Fungible": "(redacted)",
        },
        "id": {
          "Concrete": {
            "interior": {
              "X2": [
                {
                  "Parachain": "(redacted)",
                },
                {
                  "GeneralKey": "0x021f3a10587a20114ea25ba1b388ee2dd4a337ce27",
                },
              ],
            },
            "parents": "(redacted)",
          },
        },
      },
      "sender": "rPizfonc8MPuEsMxufAytHjpRW7a2YrXWHKibaJXe9ZtjKx",
    },
    "method": "TransferredMultiAssets",
    "section": "xTokens",
  },
]
`;

exports[`Karura <-> Basilisk > Karura transfer USDC to basilisk > hrmp 1`] = `
[
  {
    "data": "0x000210010400010200411f0654021f3a10587a20114ea25ba1b388ee2dd4a337ce270002093d000a1300010200411f0654021f3a10587a20114ea25ba1b388ee2dd4a337ce270002093d00000d0100040001010088dc3417d5058ec4b4503e0c12ea1a0a89be200fe98922423d4334014fa6b0ee",
    "recipient": 2090,
  },
]
`;

exports[`Karura <-> Basilisk > Karura transfer USDC to basilisk > system events 1`] = `
[
  {
    "data": {
      "amount": "(redacted)",
      "currencyId": "(redacted)",
      "who": "bXjgry3bkMHToRvMRedUhtPzEUUXdhyVX2UwY8hY156tWH4YB",
    },
    "method": "Deposited",
    "section": "tokens",
  },
  {
    "data": {
      "amount": "(redacted)",
      "currencyId": "(redacted)",
      "who": "bXjgry3bkMHToRvMRedUhtPzEUUXdhyVX2UwY8hY156tWH4YB",
    },
    "method": "Endowed",
    "section": "tokens",
  },
  {
    "data": {
      "amount": "(redacted)",
      "currencyId": "(redacted)",
      "who": "bXj4uMHTyQyvNCLHKBv6ztwkPSx8tgsrxuFtAFfWDYntXtohw",
    },
    "method": "Deposited",
    "section": "tokens",
  },
  {
    "data": {
      "messageHash": "(redacted)",
      "weight": "(redacted)",
    },
    "method": "Success",
    "section": "xcmpQueue",
  },
]
`;
